### DISCUSSION TITLE
With a DUM-based app what is the callback to handle an UPDATE message

### DISCUSSION THREAD

**Discussion by alienpenguin** [resiprocate/resiprocate#327] [Q&A] [answered]
Hi all,
this must be a silly one... when remote party sends an UPDATE i see the stack automatically respond 200 OK
but i was not able to find which is the callback of invitesessionhandler that should be overridden to handle the response to the UPDATE.

thanks for any hints :)

**Comment by sgodin** [comment]
Updates are handled automatically by DUM - in base SIP they are only used for Target Refreshing or for mid-call SDP negotiation.  If the UPDATE has an SDP body you will receive the onOffer callback.  If you want to process some other body type, take a look at the comment block at the top of InviteSessionHandler.hxx that discusses the genericOfferAnswer flag.

The only other option would be to override onReadyToSend and look for 200 responses UPDATE requests, then customize as needed.  See  https://github.com/resiprocate/resiprocate/wiki/Modifying-Decorating-messages-sent

**Reply by alienpenguin** [reply]
Hi Scott, 
thanks for your answer,
but if i want to check the update of SIP headers in the incoming INVITE? i.e. an update of P-Asserted-Identity header how can i catch it if there is no sdp and the onOffer is not called?
as for the decorating outgoing message, can i "transform" the 200 into a rejection? so fare i have used the onreadytosend to add some headers before sending.

**Comment by sgodin** [comment, marked as answer]
There is currently no callback for an Update without a body.  You would need to modify DUM.

Some additional points:
-there is an DUM incoming feature (addIncomingFeature) that could be used to "see" the inbound UPDATE messages.
-I'm not 100% what the full side effects would be if you changed DUM's 200 response to an error/rejection response, but it will be missleading to the sender, as they will think and TargetRefresh (Contact header updates) have not been applied, however they will have been applied by DUM.
